src/EnergyPlus/OutputProcessor.cc

```diff
@@ -6844,33 +6863,14 @@ void UpdateMeterReporting(EnergyPlusData &state)
         varnameLen = index(Alphas(1), '[');
         if (varnameLen != std::string::npos) Alphas(1).erase(varnameLen);
 
-        WildCard = index(Alphas(1), '*');
-        if (WildCard != std::string::npos) {
-            TestLen = WildCard;
-        }
-
         ReportFreq = determineFrequency(Alphas(2));
 
-        if (WildCard == std::string::npos) {
-            Meter = UtilityRoutines::FindItem(Alphas(1), EnergyMeters);
-            if (Meter == 0) {
-                ShowWarningError(cCurrentModuleObject + ": invalid " + cAlphaFieldNames(1) + "=\"" + Alphas(1) + "\" - not found.");
-                continue;
-            }
-
-            SetInitialMeterReportingAndOutputNames(state, Meter, false, ReportFreq, false);
-
-        } else { // Wildcard input
-            NeverFound = true;
-            for (Meter = 1; Meter <= NumEnergyMeters; ++Meter) {
-                if (!UtilityRoutines::SameString(EnergyMeters(Meter).Name.substr(0, TestLen), Alphas(1).substr(0, TestLen))) continue;
-                NeverFound = false;
-
-                SetInitialMeterReportingAndOutputNames(state, Meter, false, ReportFreq, false);
-            }
-            if (NeverFound) {
-                ShowWarningError(cCurrentModuleObject + ": invalid " + cAlphaFieldNames(1) + "=\"" + Alphas(1) + "\" - not found.");
-            }
+        int meterIndex = findMeterIndexFromMeterName(Alphas(1));
+        if (meterIndex > 0) {
+            // MeterFileOnlyIndicator is false, CumulativeIndicator is false
+            SetInitialMeterReportingAndOutputNames(state, meterIndex, false, ReportFreq, false);
+        } else {
+            ShowWarningError(cCurrentModuleObject + ": invalid " + cAlphaFieldNames(1) + "=\"" + Alphas(1) + "\" - not found.");
```

Comment:
The use of the (lambda) function allows to dry up this code block and 3 others below (4 total)